### DISCUSSION TITLE
Why is the Method Execute removed?

### DISCUSSION THREAD

**Discussion by scholle** [harborsiem/WindowsRibbon#14] [Q&A]
Hi,

I used this function to do some "clicking" from sub forms. This worked well (for me).
So, when removed, what is the replacement or different way to have the function?

Thanks
Jens

**Comment by harborsiem** [comment]
Hi,

Normally the Execute function in the Ribbon control is only used by the native Ribbon interface IUICommandHandler. If one want to use this method by code, one have knowledge how to do this.
The different way for you is to cast the Ribbon class to IUICommandHandler. Then you can call the Execute method like before.

regards
Hartmut

**Comment by harborsiem** [comment]
I have looked to the WindowsRibbon code. Here I can see that the only thing the Execute method will do is to fire an ExecuteEvent for the RibbonItem (RibbonButton, etc.).
So I think for "clicking" from sub forms  it might be better to call the event method for this ribbon item directly.

Code example for a RibbonButton with commandId 20. Instance for RibbonButton is called RibbonButton20.
With Execute method:
```
((IUICommandHandler)Ribbon).Execute(20, ExecutionVerb.Execute, null, null, null);
```

With calling event method "button20Event":
```
button20Event(RibbonButton20, new ExecuteEventArgs(null, null, null));
```
